### PR TITLE
Add MTP and Gphoto2 GIO provider to handle e.g. Android phones connecting

### PR DESCRIPTION
**Pull request by cmallwitz**
Add MTP and Gphoto2 GIO provider to handle e.g. Android phones connecting on an USB cable

### CODE REVIEW

File: application/mounts.py [hunk 1]
```diff
@@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
 		mount_uri = mount.get_root().get_uri()
 		volume = mount.get_volume()
 
+		# check if mount has been added as volume already
+		mount_already_in_volume_list = False
```
Comment (MeanEYE) [inline application/mounts.py]: Why do you have this check? It's suppose to be managed already.

Comment (cmallwitz) [inline application/mounts.py]: Hi Mladen,

Compared to GIO directly mounting a SD card (where Sunflower just 
received a single "add mount" event), there are three events when my 
Android phone gets mounted:

- "add volume" event with a volume object that has the mount correctly set
- first "add mount" event with a passed mount argument with volume being 
None on it
- second "add mount" event with a passed mount argument with volume set

With the code, I wanted to make sure we don't end up with two mounts 
displayed in the "Mount Manager" window and bookmark list.

This is on a plain Ubuntu 14.04.

Does this make any sense?

Christian

Comment (MeanEYE) [inline application/mounts.py]: I think this was a bug in GIO as 14.04 is quite old. I remember fixing the same issue with duplicate mounts as I thought GIO was suppose to work this way. After they fixed this issue in GIO they created a problem for us. If possible, please test this on some newer version just to make sure we are not creating another problem.

Comment (cmallwitz) [inline application/mounts.py]: I will check it out with Debian Jessie.

Note: I will cancel my other pull request - it is getting too big and I 
think we are better off if I prepare individual self-contained feature 
'pull requests' e.g. for the text viewer fix. This should it make easier 
for you too :-)

Christian.

Comment (MeanEYE) [inline application/mounts.py]: Thanks for your consideration. I would merge it eventually just am kind of short on free time.

Comment (cmallwitz) [inline application/mounts.py]: I verified the weird behaviour ("mount-added" callback getting called 
twice) is the same in Debian Jessie as in Ubuntu 14.04 and my code 
handles this as far as I can tell.

Interestingly, I found this bug: 
https://bugzilla.gnome.org/show_bug.cgi?id=730347 which sounds quite 
similar to what I see for MTP and Gphoto2 GIO/GVFS mounts.

Christian

**Comment by MeanEYE** [comment]
Okay, let me fix the #108 and see if this code is needed. Well, the part I commended on. The rest is mergeable.

**Pull request merged**